ssas/service/admin/api.go

```diff
@@ -146,61 +142,71 @@ Responses:
 */
 func updateGroup(w http.ResponseWriter, r *http.Request) {
 	id := chi.URLParam(r, "id")
-	trackingID := ssas.RandomHexID()
-	groupEvent := ssas.Event{Op: "UpdateGroup", TrackingID: trackingID}
-
+	ssas.SetCtxEntry(r, "Op", "ListGroup")
+	logger := ssas.GetCtxLogger(r.Context())
 	defer r.Body.Close()
 	body, _ := io.ReadAll(r.Body)
 	gd := ssas.GroupData{}
 	err := json.Unmarshal(body, &gd)
 	if err != nil {
-		groupEvent.Help = fmt.Sprintf("error in request to create group; raw request: %v; error: %v", body, err.Error())
-		ssas.OperationFailed(groupEvent)
+		logger.Error("failed to marshal JSON: ", err)
 		service.JSONError(w, http.StatusBadRequest, "invalid request body", "")
 		return
 	}
 
-	groupEvent.Help = fmt.Sprintf("calling from admin.updateGroup(), raw request: %v", string(body))
-	ssas.OperationCalled(groupEvent)
+	logger.Infof("calling from admin.updateGroup(), raw request: %v", string(body))
 	g, err := ssas.UpdateGroup(r.Context(), id, gd)
 	if err != nil {
+		logger.Errorf("failed to update group; %s", err)
 		service.JSONError(w, http.StatusBadRequest, http.StatusText(http.StatusBadRequest), fmt.Sprintf("failed to update group; %s", err))
 		return
 	}
 
 	groupJSON, err := json.Marshal(g)
 	if err != nil {
-		groupEvent.Help = err.Error()
-		ssas.OperationFailed(groupEvent)
-		service.JSONError(w, http.StatusInternalServerError, "internal error", "")
+		logger.Error("failed to marshal JSON: ", err)
+		service.JSONError(w, http.StatusInternalServerError, http.StatusText(http.StatusInternalServerError), "")
 	}
 
 	w.Header().Set("Content-Type", "application/json")
 	w.WriteHeader(http.StatusOK)
 	_, err = w.Write(groupJSON)
 	if err != nil {
-		groupEvent.Help = err.Error()
-		ssas.OperationFailed(groupEvent)
-		service.JSONError(w, http.StatusInternalServerError, "internal error", "")
+		logger.Error("failed to write response: ", err)
+		service.JSONError(w, http.StatusInternalServerError, http.StatusText(http.StatusInternalServerError), "")
 	}
 }
 
 func getSystem(w http.ResponseWriter, r *http.Request) {
 	id := chi.URLParam(r, "id")
-	trackingID := ssas.RandomHexID()
-	systemEvent := ssas.Event{Op: "UpdateSystem", TrackingID: trackingID}
+	ssas.SetCtxEntry(r, "Op", "GetSystem")
+	logger := ssas.GetCtxLogger(r.Context())
 
 	s, err := ssas.GetSystemByID(r.Context(), id)
 	if err != nil {
-		systemEvent.Help = fmt.Sprintf("; could not find system %s", id)
-		ssas.OperationFailed(systemEvent)
-		service.JSONError(w, http.StatusNotFound, "", fmt.Sprintf("could not find system %s", id))
+		logger.Errorf("could not find system %s", id)
+		service.JSONError(w, http.StatusNotFound, http.StatusText(http.StatusNotFound), fmt.Sprintf("could not find system %s", id))
 		return
 	}
 
-	ips, _ := s.GetIPsData(r.Context())
-	cts, _ := s.GetClientTokens(r.Context(), trackingID)
-	eks, _ := s.GetEncryptionKeys(r.Context(), trackingID)
+	ips, err := s.GetIPsData(r.Context())
```

Comment:
good catch on the unhandled errors, wonder if some of these are the silent ones we've seen in periods of high load.